core/variant/variant.h

```diff
@@ -836,6 +836,19 @@ struct StringLikeVariantComparator {
 	static bool compare(const Variant &p_lhs, const Variant &p_rhs);
 };
 
+struct StringLikeVariantOrder {
+	static _ALWAYS_INLINE_ bool compare(const Variant &p_lhs, const Variant &p_rhs) {
+		if (p_lhs.is_string() && p_rhs.is_string()) {
+			return p_lhs.operator String() < p_rhs.operator String();
```

Comment:
Allocating memory, filling it, then doing a comparison on the copy before dropping it immediately on the floor feels very slapdash. The memory should already be meaningful and available to use for such a comparison.

Perhaps this can be reused?:
https://github.com/godotengine/godot/blob/f032af74536b317b23c7fca3bc7318ced5537344/core/string/ustring.h#L534

Reply:
It doesn't allocate memory except when the `StringName` is created via a c-strinf, which is rare comparatively

Reply:
I'll take it, thank you!

Reply:
But will keep in mind for a future helper allowing comparison with both `String` and `StringName` generally using this correctly!